src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Rendering/ElementReferenceCapture.ts

```diff
@@ -1,12 +1,22 @@
-﻿export function applyCaptureIdToElement(element: Element, referenceCaptureId: number) {
+export function applyCaptureIdToElement(element: Element, referenceCaptureId: number) {
   element.setAttribute(getCaptureIdAttributeName(referenceCaptureId), '');
 }
 
-export function getElementByCaptureId(referenceCaptureId: number) {
+function getElementByCaptureId(referenceCaptureId: number) {
   const selector = `[${getCaptureIdAttributeName(referenceCaptureId)}]`;
   return document.querySelector(selector);
 }
 
 function getCaptureIdAttributeName(referenceCaptureId: number) {
   return `_bl_${referenceCaptureId}`;
-}
\ No newline at end of file
+}
+
+// Support receiving ElementRef instances as args in interop calls
+const elementRefKey = '_blazorElementRef'; // Keep in sync with ElementRef.cs
+DotNet.attachReviver((key, value) => {
+  if (value && typeof value === 'object' && value.hasOwnProperty(elementRefKey) && typeof value[elementRefKey] === 'number') {
+    return getElementByCaptureId(value[elementRefKey]);
```

Comment:
Should this be looking at the `key`? here? I suppose it won't be a problem because the top-value object is always an array?

Reply:
If I understand your question correctly then no, it shouldn't be looking at `key`. If we receive the following:

```
{"someElem":{"_blazorElementRef":123}}
```

Then we'd receive a call to the reviver with `key`=`"someElem"` and `value`=`{_blazorElementRef:123}`. The `key` doesn't matter; we just need to check that the `value` looks like an ElementRef, and return an `HTMLElement` instance obtained by reading the `_blazorElementRef` property from `value`.

Reply:
Ah ok, you are correct. Thanks for clarifying